prep example data from memmaped data

In [4]:
from utils.inference_utils import load_observed, load_data_non_memmaped, load_names
import os
import numpy as np

sample_data_path = '/data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/info.txt' 
celltypes_path = '/data/nchand/ImmGen/mouse/BPprofiles1000/ImmGenATAC1219.peak_matched_in_sorted.sl10004sh-4.celltypes.npy'



dataset_types = ["train", "test", "validation"]
data_names = ["bias",  "onehot",  "bp_counts", "total_counts", "peak_names"]
save_path = 'example_data/'
for split in dataset_types:
    for data_name in data_names:
        data = load_observed(sample_data_path, split, data_name=data_name)
        print(data.shape)
        n=len(data)
        new_n = n // 12
        data = data[:new_n]
        print(data.shape)
        file_name = split + '_' + data_name + '.npy'
        path = os.path.join(save_path, file_name)
        print(path)
        np.save(path, data)
        

file name /data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/train.bias.dat
(3500, 1, 998)
(291, 1, 998)
example_data/train_bias.npy
file name /data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/train.onehot.dat
(3500, 998, 4)
(291, 998, 4)
example_data/train_onehot.npy
file name /data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/train.bp_counts.dat
(3500, 90, 998)
(291, 90, 998)
example_data/train_bp_counts.npy
file name /data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/train.total_counts.dat
(3500, 90)
(291, 90)
example_data/train_total_counts.npy
file name /data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/train.names.dat
(3500,)
(291,)
example_data/train_pea

# Train bpAITAC on example data 

Note: unlike full scale data, the example data is not memmory mapped. 

./train_script_bias_spec.sh chelan 5 -1 300 BPcm_250 BP200_sample sample 20 0.001 CompositeLossBalanced 1 none 1 'Warmup' "{\"warmup_steps\": 1000}" True  complete_shallow_deprotinated_bias_quantile_normalized_4.1.25/memmap/info.txt 998
Using default celltype file: /data/nchand/ImmGen/mouse/BPprofiles1000/ImmGenATAC1219.peak_matched_in_sorted.sl10004sh-4.celltypes.npy
Sequence length is 998, setting off_by_two to true
.50000000000000000000
trial type is sample
logs/BP200_sample_L-1_5/sample/06-14-2025.08.45.txt
loss function: CompositeLossBalanced
number of filters: 300
bin size: 1 with bin type none
n scalar head layers: 1
Learning Rate Scheduler: Warmup
Learning Rate Scheduler Arguments: {"warmup_steps": 1000}
OCR evaluation: False
Set seed: True
Patience: 10
before if statement


`./train_script_bias_spec.sh chelan 5 -1 300 BPcm_250 BP200_sample sample 20 0.001 CompositeLossBalanced 1 none 1 'Warmup' "{\"warmup_steps\": 1000}" True  complete_shallow_deprotinated_bias_quantile_normalized_4.1.25/memmap/info.txt 998`


`['train.py', '/data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/info.txt', '/data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/sample_random5000_allcell_bias_corrected_normalized_5.28.23/memmap/cell_names.npy', '998', 'BP200_sample_L-1_5', 'BPcm_250', '/homes/gws/nchand/MostafaviLab/results/BPcm_250/BP200_sample_L-1_5/sample', 'CompositeLossBalanced', '200', '.50000000000000000000', 'True', '300', 'False', '1', 'none', '1', '20', '0.001', 'Warmup', '{"warmup_steps": 1000}', 'True', 'True']`

`train.py data_config_filepath cellnames_array_path input_sequence_length, model_saving_name, model_architecture, save_directory, loss_fxn, n_epochs, `


`python train.py data_config_filepath cellnames_array_path input_sequence_length model_saving_name model_architecture output_dir loss_fxn num_epochs lambda use_bias n_filters $ocr_eval $bin_size $bin_pooling_type $scalar_head_fc_layers $batch_size $learning_rate $lr_scheduler_name "$lr_scheduler_args" $set_seed $memmaped_data --off_by_two=$off_by_two --patience=$patience #> $error_dir 2>&1 `


`'train.py', '--info_file', '/data/nchand/ImmGen/mouse/BPprofiles1000/memmaped/complete_shallow_deprotinated_bias_quantile_normalized_4.1.25/memmap/info.txt', '--celltypes_path', '/data/nchand/ImmGen/mouse/BPprofiles1000/ImmGenATAC1219.peak_matched_in_sorted.sl10004sh-4.celltypes.npy', '--seq_len', '998', '--name', 'BP203_no_seed_L-1_5', '--model_name', 'BPcm_250', '--output_path', '/homes/gws/nchand/MostafaviLab/results/BPcm_250/BP203_no_seed_L-1_5/complete', '--loss_fxn', 'PoissonLoss', '--num_epochs', '200', '--_lambda', '.50000000000000000000', '--bias', 'True', '--n_filters', '300', '--ocr_eval', 'False', '--bin_size', '1', '--bin_pooling_type', 'none', '--scalar_head_fc_layers', '1', '--batch_size', '20', '--learning_rate', '0.001', '--lr_scheduler_name', 'Warmup', '--lr_scheduler_args', '{"warmup_steps": 1000}', '--set_fixed_seed', 'False', '--memmaped_data', 'True', '--off_by_two=True', '--patience=10'`

`python train.py --info_file example_data/data_config.yaml --celltypes_path  example_data/cell_names.npy --name bpAITAC_example --model_name bpAITAC --output_path example_data --seq_len 998 --memmaped_data False --num_epochs=10`

A note on model naming: during development, `bpAITAC` was refered to as `BPcm_250`. You may still see notes using `BPcm_250` instead of `bpAITAC`. These are the same things.